core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -609,8 +609,12 @@ open class TileInfo {
             tileMap.getTilesAtDistance(position, distance)
 
     fun getDefensiveBonus(): Float {
-        var bonus = 0F
-        getAllTerrains().forEach{bonus += it.defenceBonus}
+        var bonus = baseTerrainObject.defenceBonus
+        if (terrainFeatureObjects.isNotEmpty()) {
+            val otherTerrainBonus = terrainFeatureObjects.maxOf { it.defenceBonus }
+            if (abs(otherTerrainBonus) > 0.000001f) bonus = otherTerrainBonus // never compare to 0f
```

Comment:
Why not compare to 0f?  Did you see problems with this?

Reply:
We had [a discussion about float rounding and what Epsilon to use some time ago](https://github.com/yairm210/Unciv/issues?q=epsilon) (tldr: 1f.ulp might be a good one. Float.MIN_VALUE isn't). However, if the aggregations before the comparison do not include actual math, it's pointless - I see max and abs here, neither would ever round.

Reply:
More just paranoia about float comparison. I can switch to 0f for simplicity